### PR TITLE
Udev settle warning

### PR DESCRIPTION
**Pull request by keszybz**
Prompoted by the discussion on fedora-devel. We didn't actually say that udev settle should not be used anywhere. Suggestions to wording are very much welcome.

### CODE REVIEW

File: src/udev/udevadm-settle.c [hunk 1]
```diff
@@ -79,6 +84,60 @@ static int parse_argv(int argc, char *argv[]) {
...
 }
 
+static int emit_deprecation_warning(void) {
+        _cleanup_(sd_bus_flush_close_unrefp) sd_bus *bus = NULL;
+        _cleanup_free_ char *unit_path = NULL;
+        _cleanup_strv_free_ char **a = NULL, **b = NULL;
+        int r;
+
+        if (!streq_ptr(getenv("SYSTEMD_SERVICE"), "systemd-udev-settle.service"))
+                return 0;
```
Comment (poettering) [inline src/udev/udevadm-settle.c:96]:
```
_cleanup_free_ char *unit = NULL;

r = sd_pid_get_unit(0, &unit);
if (r < 0 || !streq(unit, "systemd-udev-settle.service"))
        return 0;
```

File: units/systemd-udev-settle.service.in [hunk 2]
```diff
@@ -24,4 +24,5 @@ ConditionPathIsReadWrite=/sys
 Type=oneshot
 TimeoutSec=180
 RemainAfterExit=yes
+Environment=SYSTEMD_SERVICE=%n
```
Comment (poettering) [inline units/systemd-udev-settle.service.in]: I think the sd_pid_get_unit() approach mentioned above is nicer. That said we really should look into passing a SYSTEMD_UNIT= env var to all processes we fork off possibly, but probably in a different PR.

**Comment by keszybz** [comment]
Updated with the suggested change.

bionic-amd64 failed in TEST-30. I opened https://github.com/systemd/systemd/issues/12268 to track this.

**Comment by keszybz** [comment]
I wonder if we should make the check more precise, and only warn about services which are part of dependencies of the boot unit. Using this for stuff that doesn't delay boot should be OK. For example, someone might define a service that is before `boot-complete.target` but after `systemd-udev-settle.service`. There is nothing like this atm, so it's a bit theoretical.

**Comment by keszybz** [comment]
bionic-amd64: `FAILED TESTS:  test/TEST-30-ONCLOCKCHANGE`, I'll restart CI.

bionic-arm64: fails in boot-and-services, `FAILED (failures=1, skipped=2)`. It seems to be `FAIL: test_no_options (__main__.JournalTest)`. Oh, `AssertionError: Regex didn't match: b'kernel:.*[cC]ommand line:' not found in b'`, that's a known flake. I'll restart this one too.

**Pull request merged**